src/targets/gpu/device_name.cpp

```diff
@@ -52,13 +52,15 @@ std::string get_device_name()
 bool gfx_has_fp8fnuz_intrinsics()
 {
     const auto device_name = trim(split_string(get_device_name(), ':').front());
-    return (starts_with(device_name, "gfx9") and device_name >= "gfx940");
+    return (starts_with(device_name, "gfx94"));
 }
 
 bool gfx_has_fp8ocp_intrinsics()
 {
     const auto device_name = trim(split_string(get_device_name(), ':').front());
-    return (starts_with(device_name, "gfx12") and device_name >= "gfx1200");
+    bool is_navi4x         = starts_with(device_name, "gfx12") and device_name >= "gfx1200";
+    bool is_mi35x          = starts_with(device_name, "gfx9") and device_name >= "gfx950";
```

Comment:
I get what you are trying to do here, just be mindful of future work.  If there is ever a mi36x the next person wouldn't need to make a code change here since the gfx number "should" increase in number. Yet the variable being named mi35x would still cause someone to come look at the code and have to think about the intention.  `is_mi_fp8ocp`  may show intent with less head scratching for the next developer